internal/models/serializable_int.go

```diff
@@ -0,0 +1,27 @@
+package models
+
+import "strconv"
+
+type SerializableInt int
+
+func (s SerializableInt) MarshalBinary() (data []byte, err error) {
+	return s.MarshalText()
+}
+
+func (s *SerializableInt) UnmarshalBinary(data []byte) error {
+	return s.UnmarshalText(data)
+}
```

Comment:
@Panaetius maybe I misunderstand what you mean here or I just don't know about some package you have in mind. But BinaryUnmarshaller and BinaryMarshaller are interfaces (see [here](https://pkg.go.dev/encoding#BinaryMarshaler)). And I have to implement them here so that I can store stuff in redis. In this case rather than trying to find some fancy binary representation of the int I just dump it into text. I will swap it with json.Marshal and json.Unmarshal. I suspect they also just dump the int to a string.